instance/models/instance.py

```diff
@@ -218,3 +223,14 @@ def delete(self, *args, **kwargs):
         if not kwargs.pop('ref_already_deleted', False):
             self.ref.delete(instance_already_deleted=True)
         super().delete(*args, **kwargs)
+
+
+class InstanceTag(ValidateModelMixin, models.Model):
+    """
+    Custom tags that can be applied to instances, for filtering and annotation.
+    """
+    name = models.SlugField(max_length=100, blank=False, unique=True)
```

Comment:
@smarnach Nit: Unless `SlugField` has a different [default value](https://docs.djangoproject.com/en/1.10/ref/models/fields/#blank) for `blank`, you can remove `blank=False`. (The [migration code](https://github.com/open-craft/opencraft/pull/183/files#diff-a6517275f4d31d4b15a295862ee21a86R20) doesn't have it, so I don't think the default is different.)